src/v2/modules/companies/controller.js

```diff
@@ -24,9 +30,10 @@ const getCompany = async request => {
 
 const postCompanyAddress = async (request, h) => {
   const { companyId } = request.params;
-  const { addressId, isTest, ...data } = request.payload;
+  const { addressId, isTest, roleName, ...data } = request.payload;
   try {
-    const createdEntity = await companiesService.addAddress(companyId, addressId, data, isTest);
+    const roleId = await getRoleId(roleName);
+    const createdEntity = await companiesService.addAddress(companyId, addressId, { ...data, roleId }, isTest);
```

Comment:
I'm just wondering what happens here if the roleId is not found.  It should probably respond with a `422` unprocessable entity, but I'm worried we may get a 500 error instead.